aws/internal/service/servicecatalog/id.go

```diff
@@ -32,3 +32,15 @@ func ProductPortfolioAssociationParseID(id string) (string, string, string, erro
 func ProductPortfolioAssociationCreateID(acceptLanguage, portfolioID, productID string) string {
 	return strings.Join([]string{acceptLanguage, portfolioID, productID}, ":")
 }
+
+func ProvisioningArtifactID(artifactID, productID string) string {
+	return strings.Join([]string{artifactID, productID}, ":")
+}
+
+func ProvisioningArtifactParseID(id string) (string, string, error) {
+	parts := strings.Split(id, ":")
+	if len(parts) != 2 {
```

Comment:
```suggestion
	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
```

I've seen we usually catch the emptry string as well since a user could technically input something like "example-id:" and that would pass this check